### PR TITLE
Optimize service frontend map delete: by using existing service addre…

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
…ss rather than iterating all the fronend map

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controller/workload/workload_processor.go [hunk 1]
```diff
@@ -188,15 +188,28 @@ func (p *Processor) removeWorkloadFromBpfMap(uid string) error {
...
+			nets.CopyIpByteFromSlice(&fk.Ip, networkAddress.Address)
+			if err = p.bpf.FrontendDelete(&fk); err != nil {
+				log.Errorf("delete service %s frontend key %v, err: %v", service.ResourceName(), fk, err)
+			}
+		}
+		return nil
+	}
+
+	// Otherwise fall back to iterating over the map, this can only occur on restart
 	if fks := p.bpf.FrontendIterFindKey(id); len(fks) != 0 {
```
Comment (nlgwcy) [inline pkg/controller/workload/workload_processor.go:209]: What kind of scenarios will go to this branch?

File: pkg/controller/workload/workload_processor.go [hunk 2]
```diff
@@ -208,53 +221,44 @@ func (p *Processor) deleteFrontendData(id uint32) error {
...
 func (p *Processor) removeServiceResource(resources []string) error {
-	var err error
 	for _, name := range resources {
 		telemetry.DeleteServiceMetric(name)
-		p.ServiceCache.DeleteService(name)
-		if err = p.removeServiceResourceFromBpfMap(name); err != nil {
-			return err
-		}
+		svc := p.ServiceCache.DeleteService(name)
+		_ = p.removeServiceResourceFromBpfMap(svc, name)
```
Comment (nlgwcy) [inline pkg/controller/workload/workload_processor.go:227]:
```c
_ = p.removeServiceResourceFromBpfMap(svc, name)
svc := p.ServiceCache.DeleteService(name)
```
If the sequence is changed, does DeleteService not need to be modified?

File: pkg/controller/workload/workload_processor.go (hunk 1 above)
Comment (nlgwcy) [reply pkg/controller/workload/workload_processor.go:209]: ok, I got it, `p.compareWorkloadAndServiceWithHashName` will trigger this branch.

Comment (hzxuzhonghu) [reply pkg/controller/workload/workload_processor.go:209]: Right

File: pkg/controller/workload/workload_processor.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/controller/workload/workload_processor.go:227]: no sequence is changed

File: pkg/controller/workload/workload_processor.go (hunk 1 above)
Comment (nlgwcy) [reply pkg/controller/workload/workload_processor.go:209]: But the name of this function is not good...

File: pkg/controller/workload/workload_processor_test.go [hunk 3]
```diff
@@ -253,6 +249,16 @@ func checkFrontEndMap(t *testing.T, ip []byte, p *Processor) (upstreamId uint32)
 	return
 }
 
+func checkNotExistInFrontEndMap(t *testing.T, ip []byte, p *Processor) {
+	var fk bpfcache.FrontendKey
+	var fv bpfcache.FrontendValue
+	nets.CopyIpByteFromSlice(&fk.Ip, ip)
+	err := p.bpf.FrontendLookup(&fk, &fv)
+	if err == nil {
+		t.Fatalf("expected not exist error")
```
Comment (LiZhenCheng9527) [inline pkg/controller/workload/workload_processor_test.go:258]: assert.Error(t, err)

**Pull request merged**